### PR TITLE
Add timeout-minutes

### LINKED ISSUE #45: Add `timeout_minutes: 10` to all github action workflow generated
A time out mechanism is really import to prevent a job to be running with no limits and consuming all the time available we have on CI.

we need to add `timeout_minutes: 10` to all the githuab actions workflow file we are generating from our template.

ref: https://github.com/arxlang/arx/blob/main/.github/workflows/release.yaml#L17

### PR DESCRIPTION
**Pull request by ayeankit**
Fixes #45 

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [ ] new feature
- [x] Maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ jobs:
   release:
     name: Release
     runs-on: ubuntu-latest
+    timeout-minutes: 20
```
Comment (xmnlab) [inline .github/workflows/release.yaml:10]:
```suggestion
    timeout-minutes: 10
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -51,6 +52,7 @@ jobs:
       run: make docs
 
     - name: GitHub Pages action
+      timeout-minutes: 10
```
Comment (xmnlab) [inline .github/workflows/release.yaml:55]:
```suggestion
```

we don't need a timeout here

**Review by xmnlab — CHANGES_REQUESTED** [review]
thanks for working on that @ayeankit !
as I mentioned on discord, the changes pointed by the issue #45 needs to be applied to the github actions jobs defined by the template!
thanks!

**Comment by xmnlab** [comment]
hi @ayeankit , just checking if you closed this PR on purpose.
also let me know if you have any questions :)

**Review by xmnlab — CHANGES_REQUESTED** [review]
@ayeankit not sure how you updated your branch, but the diff here shows changes from the main. so you probably should rebase your branch.
let me know if you need any help for rebasing it

File: {{cookiecutter.project_slug}}/.github/workflows/release.yaml [hunk 3]
```diff
@@ -7,6 +7,7 @@ jobs:
   release:
     name: Release
     runs-on: ubuntu-latest
+    timeout-minutes: 10  # Add timeout of 10 minutes to the job
```
Comment (xmnlab) [inline {{cookiecutter.project_slug}}/.github/workflows/release.yaml:10]: looks good. thanks
could you remove the comment please? 
this attribute is very self-explaining

Comment (ayeankit) [reply {{cookiecutter.project_slug}}/.github/workflows/release.yaml]: Done, Can you review it now?

**Review by xmnlab — CHANGES_REQUESTED** [review]
that looks good. thanks.
also, could you add it here as well please?
https://github.com/osl-incubator/cookiecutter-python/blob/main/%7B%7Bcookiecutter.project_slug%7D%7D/.github/workflows/main.yaml

**Comment by ayeankit** [comment]
@xmnlab , I have added time-out to both files. Please review it?

**Review by xmnlab — COMMENTED** [review]
Looks good to me! Thanks for working on that @ayeankit

**Pull request merged**